homeassistant/components/switch/fritzdect.py

```diff
@@ -0,0 +1,151 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+# Standard Fritz Box IP
+DEFAULT_HOST = 'fritz.box'
+
+ATTR_CURRENT_CONSUMPTION = 'Current Consumption'
+ATTR_CURRENT_CONSUMPTION_UNIT = 'W'
+
+ATTR_TOTAL_CONSUMPTION = 'Total Consumption'
+ATTR_TOTAL_CONSUMPTION_UNIT = 'kWh'
+
+ATTR_TEMPERATURE = 'Temperature'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Add all switches connected to Fritz Box."""
+    from fritzhome.fritz import FritzBox
+
+    host = config.get(CONF_HOST)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    # Log into Fritz Box
+    fritz = FritzBox(host, username, password)
+    fritz.login()
+
+    # Add all actors to hass
+    for actor in fritz.get_actors():
+        # Only add devices that support switching
+        if actor.has_switch:
+            data = FritzDectSwitchData(fritz, actor.actor_id)
+            add_devices([FritzDectSwitch(hass, data, actor.name)], True)
+
+
+class FritzDectSwitch(SwitchDevice):
+    """Representation of a FRITZ!DECT switch."""
+
+    def __init__(self, hass, data, name):
+        """Initialize the switch."""
+        self.units = hass.config.units
+        self.data = data
+        self._name = name
+
+    @property
+    def name(self):
+        """Return the name of the FRITZ!DECT switch, if any."""
+        return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        attrs = {}
+
+        if self.data.has_powermeter:
+            attrs[ATTR_TOTAL_CONSUMPTION] = "%.3f %s" % \
+              (float(self.data.total_consumption), ATTR_TOTAL_CONSUMPTION_UNIT)
+
+        if self.data.has_temperature:
+            attrs[ATTR_TEMPERATURE] = "%.1f %s" % \
+              (self.units.temperature(self.data.temperature, TEMP_CELSIUS),
+               self.units.temperature_unit)
+
+        return attrs
+
+    @property
+    def current_power_watt(self):
+        """Return the current power usage in Watt."""
+        try:
+            return float(self.data.current_consumption)
+        except ValueError:
+            return None
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        return self.data.state
+
+    def turn_on(self, **kwargs):
+        """Turn the switch on."""
+        actor = self.data.fritz.get_actor_by_ain(self.data.ain)
+        actor.switch_on()
+
+    def turn_off(self):
+        """Turn the switch off."""
+        actor = self.data.fritz.get_actor_by_ain(self.data.ain)
+        actor.switch_off()
+
+    def update(self):
+        """Get the latest data from the fritz box and updates the states."""
+        self.data.update()
+
+    @property
+    def state(self):
```

Comment:
This switch is able to measure the temperature and energy consumption. Currently, it will show the energy consumption as a graph in the web interface: 
<img width="367" alt="bildschirmfoto 2017-01-25 um 21 05 35" src="https://cloud.githubusercontent.com/assets/897823/22307104/12162950-e342-11e6-9b7e-56b2eb503451.png">

Is that something that should not be done at all or should I add a separate sensor for temperature and/or energy consumption?